flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DefaultSlotPoolServiceFactory.java

```diff
@@ -38,6 +38,7 @@ public DefaultSlotPoolServiceFactory(
     @Override
     @Nonnull
     public SlotPoolService createSlotPoolService(@Nonnull JobID jobId) {
-        return new SlotPoolImpl(jobId, clock, rpcTimeout, slotIdleTimeout, batchSlotTimeout);
+        return new DeclarativeSlotPoolService(
```

Comment:
Why not remove `DefaultSlotPoolServiceFactory`? This change make the `DefaultSlotPoolServiceFactory` same as `DeclarativeSlotPoolServiceFactory`.

Reply:
Yes, I think we should remove this factory as well.

Reply:
I think we can then also remove `ClusterOptions.ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT` and the test profile using `flink.tests.disable-declarative` because we effectively remove the non-declarative resource management. The test profile should be in `jobs-template.yml` under `legacy_slot_management`.

Also, `ResourceManagerGateway.requestSlot` should be removed.